### PR TITLE
chore: Refactor code to calculate total children recursively

### PR DESCRIPTION
**Pull request by b-l-i-n-d**
Added `calculateTotalChildren` to calculate all nested children

### CODE REVIEW

File: assets/react/v3/shared/components/fields/FormCategoriesInput.tsx [hunk 1]
```diff
@@ -166,8 +166,13 @@ interface BranchProps {
   isLastChild: boolean;
 }
 
+// all children nested under this category
```
Comment (ahamed) [inline assets/react/v3/shared/components/fields/FormCategoriesInput.tsx:169]:
```suggestion
// Calculate all children nested under this category
```
Maybe we don't need to add comment instead make the function name more descriptive

File: assets/react/v3/shared/components/fields/FormCategoriesInput.tsx [hunk 2]
```diff
@@ -166,8 +166,13 @@ interface BranchProps {
...
+const calculateTotalChildren = (option: CategoryWithChildren): number => {
+  return option.children.reduce((acc, child) => acc + calculateTotalChildren(child), option.children.length);
```
Comment (ahamed) [inline assets/react/v3/shared/components/fields/FormCategoriesInput.tsx:171]:
```suggestion
  return option.children.reduce((total, child) => total + calculateTotalChildren(child), option.children.length);
```

File: assets/react/v3/shared/utils/util.ts [hunk 3]
```diff
@@ -256,20 +256,20 @@ export const parseNumberOnly = (value: string) => {
...
-		if (!inThrottle) {
-			func.apply(this, args);
-			inThrottle = true;
-
-			setTimeout(() => {
-				inThrottle = false;
-			}, limit);
-		}
-	};
+  let inThrottle = false;
```
Comment (ahamed) [inline assets/react/v3/shared/utils/util.ts:259]: I don't find any usability of this variable.

**Review by ahamed — CHANGES_REQUESTED** [review]

File: assets/react/v3/shared/components/fields/FormCategoriesInput.tsx (hunk 1 above)
Comment (b-l-i-n-d) [reply assets/react/v3/shared/components/fields/FormCategoriesInput.tsx:169]: change the name to `getTotalNestedChildrenCount`

File: assets/react/v3/shared/utils/util.ts (hunk 3 above)
Comment (b-l-i-n-d) [reply assets/react/v3/shared/utils/util.ts:259]: `inThrottle` is used inside the return function

**Pull request merged**